src/v2/System/Relay/SystemQueryRenderer.tsx

```diff
@@ -28,19 +29,26 @@ export function SystemQueryRenderer<T extends OperationType>({
   lazyLoad = false,
   lazyLoadThreshold = 1500,
   placeholder,
-  environment,
+  environment: _environment,
   variables = {},
   render,
   ...rest
 }: SystemQueryRendererProps<T>): JSX.Element {
   const isMounted = useDidMount()
+
   const { isEnteredView, Waypoint } = useLazyLoadComponent({
     threshold: lazyLoadThreshold,
   })
+
   const showPlaceholder =
     debugPlaceholder || !isMounted || (lazyLoad && !isEnteredView)
 
+  const { relayEnvironment } = useSystemContext()
+
+  const environment = _environment ?? relayEnvironment
```

Comment:
Why is this now needed?

Reply:
It's passed in two places below — could inline the `??` in both otherwise

Reply:
Oh i just mean, it seems strange that `relayEnvironment` should ever be passed into `SystemQueryRenderer`, unless that code is very old. It should always be available from `<Boot>`.

Reply:
It has always needed it (every instance of `SystemQueryRenderer` explicitly passes it in, it was just never given a default before. What seems strange about this though — it's needed in `QueryRenderer`.